ui/lesson-renderer/src/main/kotlin/org/cru/godtools/tool/lesson/ui/LessonActivity.kt

```diff
@@ -193,11 +193,18 @@ class LessonActivity :
                 }
             }
 
+            val shareLinkUri by shareLinkUriLiveData.observeAsState()
+
             // render the Lesson
             ProvideRendererServices(resources = resourceFileSystem, tipsRepository = tipsRepository) {
                 GodToolsTheme(darkTheme = false) {
                     ContentWithOverlays {
-                        RenderLesson(state)
+                        RenderLesson(
+                            state = state,
+                            onShareClick = if (shareLinkUri != null && state is LessonScreen.UiState.Loaded) {
+                                { shareCurrentTool() }
+                            } else null
```

Comment:
there is already an event callback present in the `LessonScreen.UiState.Loaded` object, you should add an new share link event in there for this callback